### PR TITLE
[SEC-2948]: Add SAST scans for Security

### PR DESCRIPTION
**Pull request by karankohli-cf**
Adds Polaris SAST scaning GHA. https://contentful.atlassian.net/browse/SEC-2948

**_What will change?_**

* Stuff that is going to change
* even more interesting stuff that is going to change

## Security

_Before you click Merge, take a step back and think how someone could break the [Confidentiality, Integrity and Availability](https://docs.google.com/presentation/d/1YdFlYBLnbNoiSAMOKjopiF4u34StXTK2qYdOLkMsEKo/edit?usp=sharing) of the code you've just written. Are secrets secret? Is there any sensitive information disclosed in logs or error messages? How does your code ensure that information is accurate, complete and protected from modification? Will your code keep Contentful working and functioning?_

### CODE REVIEW

File: .github/workflows/sast.yaml [hunk 1]
```diff
@@ -0,0 +1,31 @@
+name: SAST (Static Application Security Testing)
+
+on:
+  push:
+    branches: [master, main]
```
Comment (mattvanvoorst-contentful) [inline .github/workflows/sast.yaml:5]:
```suggestion
    branches: [master, main, main-private]
```

File: .github/workflows/sast.yaml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+  pull_request:
+    branches: [master, main]
```
Comment (mattvanvoorst-contentful) [inline .github/workflows/sast.yaml:7]:
```suggestion
    branches: [master, main, main-private]
```

**Review by mattvanvoorst-contentful — CHANGES_REQUESTED** [review]
This repository uses two main branches. One that gets forked by our users and one that powers our preview environments. We'd want to run this on both

**Review by mattvanvoorst-contentful — APPROVED** [review]
Thanks!

**Comment by mattvanvoorst-contentful** [comment]
It's been approved, but it's not progressing. I'll also poke the team for an additional approval

**Comment by mattvanvoorst-contentful** [comment]
Two different people have approved the Vercel flow, but it doesn't seem to want to progress. It did work for the other two repo's. We'll give it a bit more time and try again in half an hour

**Review by cf-remylenoir — APPROVED** [review]

**Comment by mattvanvoorst-contentful** [comment]
We can merge it, but the Vercel authorisation step doesn't seem to want to progress. We can merge it regardless and see if it'll resolve itself

**Comment by mattvanvoorst-contentful** [comment]
@cf-remylenoir what do you think, merge it?

**Pull request merged**